### ISSUE TITLE
regression: 2.12.x release-package-windows keeps hanging

### ISSUE THREAD

**Issue by SethTisue** [scala/scala-dev#80]
so e.g. https://scala-ci.typesafe.com/job/scala-2.12.x-release-package-windows/150/console

```
[error] Reference to undefined setting: 
[error] 
[error]   windows:packageBin from windows:packageBin (Y:\jenkins\workspace\scala-2.12.x-release-package-windows\build.sbt:24)
[error]  
[error] Use 'last' for the full log.
Project loading failed: (r)etry, (q)uit, (l)ast, or (i)gnore?
```

looks like sbt is hanging waiting for input.

it isn't intermittent, several consecutive runs hung and had to be manually aborted. (but since Jan 23, there haven't been further runs, pending resolution of #77)
- last green run: https://scala-ci.typesafe.com/job/scala-2.12.x-release-package-windows/148/console  
  scala rev: https://github.com/scala/scala/commit/93fa559f62637fc3a49d7b961a1d07cf843b0e6b  
  scala-dist rev: https://github.com/scala/scala-dist/commit/b2fccc985be48b4f4cc26ba6f2627818393daf03
- first aborted run: https://scala-ci.typesafe.com/job/scala-2.12.x-release-package-windows/149/consoleFull  
  scala rev: https://github.com/scala/scala/commit/a45a9abb6ec66dad873a57196ca8851bfadb5557  
  scala-dist rev: https://github.com/scala/scala-dist/commit/b8b12718a60caee124aeccf7f77f631350047b51

**Comment by SethTisue** [comment]
hmm, I may just need to merge the latest 2.11.x fixes in the scala-dist repo (addressing regressions due to the sbt-native-packager upgrade) onto 2.12.x. I'll try that next.

**Comment by SethTisue** [comment]
project/ScalaDist.scala has:

``` scala
    if (sys.props("os.name").toLowerCase(java.util.Locale.US) contains "windows")
      Wix.settings :+ (mappings in upload += uploadMapping(packageBin in Windows).value)
```

project/Wix.scala customizes the Windows artifact name as follows:

``` scala
    packageBin in Windows := {
      val versioned = target.value / s"${name.value}-${version.value}.msi"

      IO.copyFile((packageBin in Windows).value, versioned)
      versioned
    }
```

perhaps we should customize at upload time instead, a la https://github.com/scala/scala-dist/commit/2d525f867f7839be582c36d2a6d92ee194287447? (still puzzled, though, why we would be forced to change approaches here, but I do think customizing at upload time is actually a cleaner solution anyway)

**Comment by SethTisue** [comment]
so in https://github.com/scala/scala-dist/pull/160 I merged 2.11.x onto 2.12.x in the scala-dist repo, and it appears from the successful dry run at https://scala-ci.typesafe.com/job/scala-2.12.x-release-package-windows/155/ that the problem may be fixed. let's wait and see what happens tonight.

**Comment by SethTisue** [comment]
two greens in a row, so I figure this must be fixed.
